### DISCUSSION TITLE
Cannot setup webhook trigger

### DISCUSSION THREAD

**Discussion by mroziken** [Budibase/budibase#5009] [Help] [answered]
I'm trying to setup a webhook trigger. 

1. The first problem is that I cannot find on budibase the documentation for it. The link "Learn about webhhook" from Webhook setup screen leads to https://docs.budibase.com/automate/steps/triggers which results in 404 error "Page not found" - Can anybody provide me the proper link ?

2. The setup webhook screen looks like that

![image](https://user-images.githubusercontent.com/9000976/159528450-4d647950-24ae-4a39-91b0-8a1b88745410.png)

Is it normal that URL starts from api/webhook/schema  instead of https://<myapp>.budibase.app ?

3. When I send the POST request to https://<myapp>.budibase.app/<URL from Setup Webhook> screen I got successfull response and (statusCode 200)  and body looks like:
{
    "ok": true,
    "id": "<some string>",
    "rev": "<some_string>"
}

Looks positive but I do not see acction being triggered

Can somebody help me to setup find out what I am doing wrong?

**Comment by melohagan** [comment, marked as answer]
Hi @mroziken 

The base URL should be included in your API calls, so it will take the form: `https://<domain>.budibase.app/api/webhook/schema/...`

The 200 success means that the schema has been successfully created. You will now need to use the **Trigger URL** as your webhook.

It will be the second link presented in the automation webhook setup:

![Screenshot 2022-03-22 at 16 45 52](https://user-images.githubusercontent.com/101575380/159531824-1ecb2bf3-1481-4a9d-801c-1ad0c6563539.png)

Also note that automations only apply to published apps. 

I'll enquire if the domain can be included in the quick copy link, and where the documentation has gone.

Let me know how it goes!

**Comment by mroziken** [comment]
@melohagan Thank you for your response.

The when I sent post request to the Trigger URL
I receive again statusCode  200 and body is:

{
    "message": "Webhook trigger fired successfully"
}

But action is not triggered. The action is add row to a table. It works when in test mode I put the body of webhook in webhook JSON input field but it does not work when I send request from postman. The application is published before sending request from postman. Is there a way sto see kind of log of webhook receiver?

**Comment by mroziken** [comment]
Ok.. it looks like it works. However I observed a strange thing. In order to see the record in my app or in builder table view I must do the folloiwng steps:

1. Run webhook from postman
2. Logout and login to budibase portal
3. Select in builder the app
4. Go to data tab and select the table where I expect new record

Then if I go to my published app and refresh the page with list of records for the table I can see it. 

Anyway webhook works for me so I'm marking the thread as answered.

**Reply by melohagan** [reply]
Great, glad that's working!
 
I've noticed myself after receiving a successful 200 from postman on the Trigger URL, that it can take a few minutes for the automation to trigger. You shouldn't need to logout / log-in, but you would need a page refresh.

**Comment by mroziken** [comment]
Yes, I've noticed that as well that page refresh does work.

śr., 23 mar 2022 o 09:50 melohagan ***@***.***> napisał(a):

> Great, glad that's working!
>
> I've noticed myself after receiving a successful 200 from postman on the
> Trigger URL, that it can take a few minutes for the automation to trigger.
> You shouldn't need to logout / log-in, but you would need a page refresh.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/5009#discussioncomment-2420859>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACEVQEETWAKE2E2C257X7JLVBLLMBANCNFSM5RLMUB3Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Michal Mrozik

+48   608 826 701
